includes/multi-currency/Helpers/OrderMetaHelper.php

```diff
@@ -0,0 +1,443 @@
+			$old_exchange_rate,
+			$new_exchange_rate
+		);
+		$order->add_order_note( $note, 0, true );
+
+		// Update the exchange rate and the log on the order.
+		$order->update_meta_data( '_wcpay_multi_currency_order_exchange_rate', $new_exchange_rate );
+		$order->update_meta_data( '_wcpay_multi_currency_order_exchange_rate_log', $exchange_rate_log );
+		$order->save_meta_data();
+	}
+
+	/**
+	 * Maybe add the meta box.
+	 *
+	 * @param string $post_type The type of post being viewed.
+	 * @param object $post      The post object for the post being viewed.
+	 *
+	 * @return void
+	 */
+	public function maybe_add_meta_box( $post_type, $post ) {
+		// Confirm we should be working on the order, if not, exit.
+		$order = $this->confirm_actions( $post );
+		if ( ! $order || ! function_exists( '\wc_get_page_screen_id' ) ) {
+			return;
+		}
+
+		// Get the order edit screen to be able to add the meta box to.
+		$wc_screen_id = \wc_get_page_screen_id( 'shop-order' );
+
+		add_meta_box( 'wcpay_mc_order_meta_helper_meta_box', __( 'Multi-Currency Meta Helper', 'woocommerce-payments' ), [ $this, 'display_meta_box_content' ], $wc_screen_id, 'advanced', 'high' );
+	}
+
+	/**
+	 * Displays the content of the meta box.
+	 *
+	 * @param \WC_Order $order The order we are working with.
+	 *
+	 * @return void
+	 */
+	public function display_meta_box_content( $order ) {
+		// Again, make sure we are actually working with an order.
+		$order = wc_get_order( $order );
+		if ( ! $order ) {
+			return;
+		}
+
+		// Start getting all of the items we need.
+		$order_currency = $order->get_currency();
+		$intent_id      = $order->get_meta( '_intent_id' );
+		$payment_method = $order->get_payment_method();
+		$is_wcpay       = false !== strpos( $payment_method, 'woocommerce_payments' );
+
+		// Define the store items.
+		$store_items = [
+			'store_currency' => [
+				'label' => __( 'Store Currency', 'woocommerce-payments' ),
+				'value' => get_woocommerce_currency(),
+			],
+		];
+
+		// Define the order meta items.
+		$order_meta_items = [
+			'order_currency'         => [
+				'label' => __( 'Order Currency', 'woocommerce-payments' ),
+				'value' => $order_currency,
+			],
+			'order_default_currency' => [
+				'label' => __( 'Order Default Currency', 'woocommerce-payments' ),
+				'value' => $order->get_meta( '_wcpay_multi_currency_order_default_currency' ),
+			],
+			'payment_method'         => [
+				'label' => __( 'Payment Method ID', 'woocommerce-payments' ),
+				'value' => $payment_method,
+			],
+			'payment_method_title'   => [
+				'label' => __( 'Payment Method Title', 'woocommerce-payments' ),
+				'value' => $order->get_payment_method_title(),
+			],
+			'intent_id'              => [
+				'label' => __( 'Intent ID', 'woocommerce-payments' ),
+				'value' => $intent_id,
+			],
+			'intent_currency'        => [
+				'label' => __( 'Intent Currency', 'woocommerce-payments' ),
+				'value' => $order->get_meta( '_wcpay_intent_currency' ),
```

Comment:
Can we convert this to uppercase to make it look like the rest of the currencies?

Reply:
I went through and set them all to be uppercase, just to be sure. This is in https://github.com/Automattic/woocommerce-payments/pull/6582/commits/710beecd504906ed5ff896a55012d58cda775092

In my testing I never noticed any of them being lowercase, but then I did a database search and found that subscription renewals have the `_wcpay_intent_currency` as lowercase. I wonder if this affects anything else elsewhere.